149 building select list split

Heldereim opened the pull request:
Split the selectables in building select list into 4 categories; Gathering, Producing, Storing, and Utility

Selecting and placing buildings should still work the same

Heldereim committed:
Split selectables into categories

Split the selectables contained in building select list into four categories: Gathering, Producing, Storing, Utility

Heldereim committed:
Updated styleboxtexture

Updated styleboxtexture to remove weird behaviour where the tabs would move when one was selected

Heldereim committed:
Sell value resets on building clicked

The value of the selected stored resources now resets between building clicks. Also moved some code outside the single/multi if statements since they code was executed either way

linuskar reviewed (changes requested):
Can you fix these errors in ui_menu.gd:
![image](https://github.com/user-attachments/assets/a59f8de5-6c2c-4008-8c9a-0fc622120d43)

Heldereim committed:
Moved files, moved variables to new script

Moved script files into the scripts/UI folder. Made a new script for the list in building select list that extends UIMenu and moved the TabBar and BuildingsContainer variables into that script

linuskar commented:
Some conflicts to fix

Heldereim committed:
Merge branch 'main' into 149-building-select-list-split

Heldereim committed:
Fixed a scrolling issue in the list

Fixed an issue with the building select list that made scrolling the list look weird

linuskar commented:
One conflict to fix

Heldereim committed:
Merge branch 'main' into 149-building-select-list-split

linuskar reviewed (approved):
LGTM
